src/backend/gporca/libgpos/src/task/CAutoTaskProxy.cpp

```diff
@@ -131,8 +131,10 @@ CAutoTaskProxy::Create(void *(*pfunc)(void *), void *arg, BOOL *cancel)
 	// auto pointer to hold new task context
 	CAutoP<CTaskContext> task_ctxt;
 
+	CWorker *worker = CWorker::Self();
+	GPOS_ASSERT(NULL != worker);
```

Comment:
How about moving this GPOS_ASSERT to the CWorker::Self method? And at the same time, add GPOS_ASSERT to the IWorker::Self method.

Reply:
> And at the same time, add GPOS_ASSERT to the IWorker::Self method.

It has a null check:
https://github.com/arenadata/gpdb/blob/f1b7de75b0b1a8e28efb5f9f26f74dfa7e81ed70/src/backend/gporca/libgpos/src/task/IWorker.cpp#L28-L39

Should it be replaced with an assert?

Reply:
No! My bad